dm/syncer/data_validator.go

```diff
@@ -959,15 +1058,71 @@ func (v *DataValidator) OperateValidatorError(validateOp pb.ValidationErrOp, err
 	return v.persistHelper.operateError(validateOp, errID, isAll)
 }
 
-// return snapshot of the current table status.
-func (v *DataValidator) getTableStatus() map[string]*tableValidateStatus {
-	v.RLock()
-	defer v.RUnlock()
-	tblStatus := make(map[string]*tableValidateStatus)
-	for key, tblStat := range v.tableStatus {
-		stat := &tableValidateStatus{}
-		*stat = *tblStat // deep copy
-		tblStatus[key] = stat
+func (v *DataValidator) getAllErrorCount(timeout time.Duration) ([errorStateTypeCount]int64, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	tctx := tcontext.NewContext(ctx, v.L)
+	// todo: should create a separate db to get error count, since validator maybe stopped or initializing,
```

Comment:
```suggestion
	// todo: should create a separate db to get error count, since validator may be stopped or being initialized,
```

What did you mean about `a separate db`?

Reply:
db in validator will be closed on stop, and we need to support get validator status when it's stopped

Reply:
> > being initialized
> 
> it happens during initializing

`initialize` is a v.t. So we could say "initialize itself" or "being initialized".